pkg/scheduler/plugins/deviceshare/device_plugin_adapter.go

```diff
@@ -141,6 +179,74 @@ func (a *huaweiGPUDevicePluginAdapter) Adapt(object metav1.Object, allocation []
 	return nil
 }
 
+type cambriconGPUDevicePluginAdapter struct {
+	clock clock.Clock
+}
+
+func (a *cambriconGPUDevicePluginAdapter) Adapt(ctx *DevicePluginAdaptContext, object metav1.Object, allocation []*apiext.DeviceAllocation) error {
+	if len(allocation) > 1 {
+		return fmt.Errorf("multiple gpu share is not supported on device side")
+	}
+
+	core, ok := allocation[0].Resources[apiext.ResourceGPUCore]
+	if !ok {
+		return fmt.Errorf("gpu core resource is required")
+	}
+	vcore := core.Value()
+
+	memory := allocation[0].Resources[apiext.ResourceGPUMemory]
+	if memory.Value() < cambriconVMemoryUnit {
+		return fmt.Errorf("gpu memory must not be less than %d bytes", cambriconVMemoryUnit)
+	}
+	vmemory := memory.Value() / cambriconVMemoryUnit
+
+	if err := a.lockNode(ctx, ctx.client, ctx.node); err != nil {
```

Comment:
If err != nil, does it mean rescheduling?

Reply:
Yes, it will lead to a prebind error and trigger rescheduling.